add some more documentation for e2e-tests (#1194)
fermyon/spin: tests/README.md

## Important files and their function

`crates/e2e-testing`     - All the test framework/utilities that are required for `e2e-tests`<br/>
`tests/testcases/mod.rs` - All the testcase definitions should be added here.<br/>
`tests/spinup_tests.rs`  - All tests that we want to run with `Spin Up` should be added here<br/>
`tests/testcases/<dirs>` - The testcases which require corresponding `spin app` pre-created should be added here<br/>

> ## key concepts and types

Reviewer: nit: capitalize the headers.

Author: thnx, done

### [trait Controller](../crates/e2e-testing/src/controller.rs#L12)

> This defines a trait which can be implemented by different deployment models (e.g. [`spin up`](../crates/e2e-testing/src/spin_controller.rs#L15) or `Fermyon cloud`). Using this we can reuse same testcases, which can be executed against these different deployment models. e.g. they may chose to have different way to start/stop the spin apps.

Reviewer: nit: "Fermyon *C*loud"

Author: thnx Radu. done

> This defines a trait which can be implemented by different deployment models (e.g. [`spin up`](../crates/e2e-testing/src/spin_controller.rs#L15) or `Fermyon Cloud`). Using this we can reuse same testcases, which can be executed against these different deployment models. e.g. they may chose to have different way to start/stop the spin apps.

Reviewer 2: ```suggestion
This defines a trait which can be implemented by different deployment models (e.g. [`spin up`](../crates/e2e-testing/src/spin_controller.rs#L15) or `Fermyon Cloud`). Using this we can reuse the same testcases, which can be executed against these different deployment models (e.g. they may choose to have different way to start/stop the spin apps.)
```

### [TestCase](../crates/e2e-testing/src/testcase.rs#L22)

> This help us configure the different scenarios/steps which are required for running a specific test app. e.g. it has [TestCase.trigger_type](../crates/e2e-testing/src/testcase.rs#L42) which tells the type of `trigger` a particular test app uses. and similarly it has [TestCase.plugins](../crates/e2e-testing/src/testcase.rs#L53) which tell the pre-requisite [`plugins`](https://developer.fermyon.com/spin/plugin-authoring) that are required to run this test app.

Reviewer 2: ```suggestion
This helps us configure the different scenarios/steps which are required for running a specific test app. For example, [TestCase.trigger_type](../crates/e2e-testing/src/testcase.rs#L42) indicates the `trigger` a particular test app uses and [TestCase.plugins](../crates/e2e-testing/src/testcase.rs#L53) indicates which prerequisite [`plugins`](https://developer.fermyon.com/spin/plugin-authoring) are required to run this test app.
```
